struct.go

```diff
@@ -295,20 +292,34 @@ func (s *Section) mapToField(val reflect.Value, isStrict bool, sectionIndex int)
 			continue
 		}
 
-		rawName, _, allowShadow, allowNonUnique := parseTagOptions(tag)
+		rawName, _, allowShadow, allowNonUnique, extends := parseTagOptions(tag)
 		fieldName := s.parseFieldName(tpField.Name, rawName)
 		if len(fieldName) == 0 || !field.CanSet() {
 			continue
 		}
 
 		isStruct := tpField.Type.Kind() == reflect.Struct
 		isStructPtr := tpField.Type.Kind() == reflect.Ptr && tpField.Type.Elem().Kind() == reflect.Struct
-		isAnonymous := tpField.Type.Kind() == reflect.Ptr && tpField.Anonymous
-		if isAnonymous {
+		isAnonymousPtr := tpField.Type.Kind() == reflect.Ptr && tpField.Anonymous
+		if isAnonymousPtr {
 			field.Set(reflect.New(tpField.Type.Elem()))
 		}
 
-		if isAnonymous || isStruct || isStructPtr {
+		if extends && (isAnonymousPtr || (isStruct && tpField.Anonymous)) {
+			if isStructPtr && field.IsNil() {
+				field.Set(reflect.New(tpField.Type.Elem()))
+			}
+			fieldSection := s
+			if rawName != "" {
+				sectionName = s.name + s.f.options.ChildSectionDelimiter + rawName
+				if secs, err := s.f.SectionsByName(s.name + s.f.options.ChildSectionDelimiter + rawName); err == nil && len(secs) > 0 && sectionIndex < len(secs) {
```

Comment:
```suggestion
				if secs, err := s.f.SectionsByName(sectionName); err == nil && sectionIndex < len(secs) {
```

Reply:
I think `len(secs) > 0` is covered when `sectionIndex < len(secs)` is true. 

In the case of a bad `sectionIndex` (i.e. -1), [this value is not controlled by user input](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/go-ini/ini%24%4061851be+mapToField&patternType=literal) so I don't think it's a problem. At very least we can have a check to make sure `sectionIndex` is never smaller than 0 in the beginning of this method.